client/state/sites/reducer.js

```diff
@@ -135,43 +135,58 @@ export function items( state = {}, action ) {
 		case SITE_SETTINGS_UPDATE:
 		case SITE_SETTINGS_RECEIVE: {
 			const { siteId, settings } = action;
+			const site = state[ siteId ];
 
-			// A site settings update may or may not include the icon property.
-			// If not, we should simply return state unchanged.
-			if ( ! settings.hasOwnProperty( 'site_icon' ) ) {
+			if ( ! site ) {
 				return state;
 			}
 
-			const mediaId = settings.site_icon;
+			let nextSite = { ...site };
```

Comment:
Minor-ish: We're creating a shallow clone of the site even if we never end up making any changes. We could do this at the point we need it instead:

```js
let nextSite = site;

// ...

case 'blog_public':
	const isPrivate = parseInt( settings.blog_public, 10 ) !== 1;
	if ( site.is_private === isPrivate ) {
		return memo;
	}

	nextSite = { ...nextSite, is_private: isPrivate };
	break;
```

Only downside of this is that if multiple keys have new values, we'd clone the site multiple times. I think it's the much more likely scenario that nothing will have changed (thus zero cloning) than multiple changed values though.